### PR TITLE
feat(ST-149): integrate fetch submitted assignments api

### PR DESCRIPTION
**Pull request by akib-raihan-sazim**
## Description of Change
- Created SubmissionModal component
- Created interface file for the component
- Updated AssignmentCard component to include SubmissionModal
- Update assignments.api file to include the endpoint for fetching submitted assignments
- Included response dto and interface

## Associated Ticket Link(s)
- https://trello.com/c/lCzp6VLM

## Screenshots / Screen Recordings

https://github.com/user-attachments/assets/4f4062f8-e929-46b8-a9e1-383ae8ed20ca

### CODE REVIEW

File: modules/Classroom/components/SubmissionsModal/SubmissionsModal.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+import { ISubmissionsModalProps } from "./SubmissionsModal.interface";
+
+const SubmissionsModal: React.FC<ISubmissionsModalProps> = ({
+  opened,
+  onClose,
+  assignmentId,
+  dueDate,
+}) => {
+  const { data: submissions, isLoading, isError } = useGetAssignmentSubmissionsQuery(assignmentId);
```
Comment (ishmam-iqbal-sazim) [inline modules/Classroom/components/SubmissionsModal/SubmissionsModal.tsx:25]: Query should be skipped if assignmentId is falsy

File: shared/redux/rtk-apis/assignments/assignments.api.ts [hunk 2]
```diff
@@ -73,6 +78,20 @@ const assignmentsApi = projectApi.injectEndpoints({
...
+        method: "GET",
+      }),
+      transformResponse: (response: IApiSubmissionResponse[]): StudentResponseDto[] =>
+        response.map((submission) => ({
+          firstName: submission.student.user.firstName.trim(),
+          lastName: submission.student.user.lastName.trim(),
+          fileUrl: submission.fileUrl,
+          createdAt: submission.createdAt,
+        })),
+    }),
```
Comment (ishmam-iqbal-sazim) [inline shared/redux/rtk-apis/assignments/assignments.api.ts:94]: Don't we need cache invalidation here?

Comment (akib-raihan-sazim) [reply shared/redux/rtk-apis/assignments/assignments.api.ts]: Bhaiya, as I am fetching submissions, no need for invalidation now.

File: shared/redux/rtk-apis/assignments/assignments.api.ts [hunk 3]
```diff
@@ -76,6 +81,23 @@ const assignmentsApi = projectApi.injectEndpoints({
...
```
Comment (ishmam-iqbal-sazim) [inline shared/redux/rtk-apis/assignments/assignments.api.ts:100]: This should be inside `submissions/submission.api.ts`

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Approved pending comments

File: modules/Classroom/components/AssignmentCard/AssignmentCard.tsx [hunk 4]
```diff
@@ -20,6 +21,7 @@ const AssignmentCard: React.FC<IAssignmentCardProps> = ({ assignment, classroomI
   const [editModalOpened, setEditModalOpened] = useState(false);
   const [deleteModalOpened, setDeleteModalOpened] = useState(false);
   const [submitModalOpened, setSubmitModalOpened] = useState(false);
+  const [submissionsModalOpened, setSubmissionsModalOpened] = useState(false);
```
[the remark this answers is not in the document]
Comment (ragib-hossain-sazim) [reply modules/Classroom/components/AssignmentCard/AssignmentCard.tsx:24]: [Super Nitpick] let's use `isSubmissionModalOpen` singular.

**Review by ragib-hossain-sazim — APPROVED** [review]
Approved pending comments

**Pull request merged**